src/compiler/scala/tools/nsc/transform/Delambdafy.scala

```diff
@@ -306,7 +306,7 @@ abstract class Delambdafy extends Transform with TypingTransformers with ast.Tre
       case dd: DefDef if dd.symbol.isLiftedMethod && !dd.symbol.isDelambdafyTarget =>
         // scala/bug#9390 emit lifted methods that don't require a `this` reference as STATIC
         // delambdafy targets are excluded as they are made static by `transformFunction`.
-        if (!dd.symbol.hasFlag(STATIC) && !methodReferencesThis(dd.symbol)) {
+        if (!dd.symbol.hasFlag(STATIC) && !dd.symbol.hasFlag(SYNCHRONIZED) && !methodReferencesThis(dd.symbol)) {
```

Comment:
```suggestion
        // a synchronized method cannot be static (`methodReferencesThis` will not see the implicit this reference due to `this.synchronized`)
        if (!dd.symbol.hasFlag(STATIC | SYNCHRONIZED) && !methodReferencesThis(dd.symbol)) {
```

Reply:
Looks good! I suggested a small tweak to exclude both flags in one check (hasFlag checks whether any bit in the mask is set), and added a small comment.